### DISCUSSION TITLE
FreeRDP certificate store?

### DISCUSSION THREAD

**Discussion by veloras** [FreeRDP/FreeRDP#7016] [Q&A]
I'm new to FreeRDP and I'm using v2.1.1 with Apache Guacamole v1.3.0 on RHEL8 for remote RDP access.  RDP connections work as expected if I use the "ignore-cert=true" parameter. I'd like to eliminate the need for "ignore-cert=true" if I can.  Can anyone provide guidance on how to get rid of that?  The servers to which I am connecting have valid certs signed by our internal CA and I see our CA listed when I run "trust list".
Which certificate store does FreeRDP use?
Thanks for any assistance.
-veloras

**Comment by akallabeth** [comment]
FreeRDP uses a 2step system:
1. If the supplied `/cert:hash:xxxxxx` matches
1. Try to validate with OpenSSL
2. Fall back to `~/.config/freerdp/known_hosts2` and check if a manually accepted certificate hash matches for the desired destination.

Limitations:
1. The trust root certificate must be installed and work with `openssl` on the client machine
1. This will only work if the certificate `common name` matches the hostname connecting to.

**Reply by veloras** [reply]
akallabeth, thanks for your response.  I'm not seeing the freerdp folder in my ~/.config folder.  I installed freerdp using yum.
Is there a different location where I might find /freerdp/known_hosts2?
Where do I find the `/cert:hash:xxxxxx folder/file`?
Thank you again for your help.
-veloras

**Comment by akallabeth** [comment]
@veloras 
1. `/cert:hash:<somehash>` is a command line option to directly tell which certificate hash you expect (see command line help output)
2. The file does not exist unless you have manually accepted some certificate
